### PR TITLE
Pub Tests

### PR DESCRIPTION
**Pull request by maxxfrazer**
- Using `dart-package-analyzer` to get pub score.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -20,16 +20,21 @@ jobs:
...
-      - run: flutter pub get
-      - run: bash ci/dartfmt_check.sh
-      - run: flutter analyze
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
+      - name: Check scores
+        env:
+          TOTAL: ${{ steps.analysis.outputs.total }}
+          TOTAL_MAX: ${{ steps.analysis.outputs.total_max }}
+        run: |
+          if (( $TOTAL < 100 ))
```
Comment (tadaspetra) [inline .github/workflows/build.yml:33]: Version `4.2.1` was at 120 pub points. I think we should have the minimum be 120. Package points shouldn't be going down.

**Review by tadaspetra — CHANGES_REQUESTED** [review]

File: lib/analysis_options.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+analyzer:
```
Comment (tadaspetra) [inline lib/analysis_options.yaml:1]: Although this works, there shouldn't be an `analysis_options.yaml` in the `lib` folder, but not sure why the main `analysis_options.yaml` isn't catching the `classes.g.dart` file

**Review by tadaspetra — CHANGES_REQUESTED** [review]

File: lib/analysis_options.yaml (hunk 2 above)
Comment (littleGnAl) [reply lib/analysis_options.yaml:1]: @maxxfrazer @tadaspetra It's a bug of `pana`, the better way is to add a `build.yaml` file to add the `// ignore_for_file` for the `*.g.dart` files.
```yaml
targets:
  $default:
    builders:
      source_gen|combining_builder:
        options:
          ignore_for_file:
          - non_constant_identifier_names
          - deprecated_member_use_from_same_package
          - unused_element
```

see: https://github.com/dart-lang/pana/issues/1013#issuecomment-1034703342

**Comment by littleGnAl** [comment]
The integration tests will fail, cause the forked repository can not access the `screct` of the github action, we are still investigating how to solve it.

**Comment by littleGnAl** [comment]
@maxxfrazer I will fix the analysis in this PR https://github.com/AgoraIO/Agora-Flutter-SDK/pull/620. You can just keep the github action part for this PR.

**Comment by maxxfrazer** [comment]
@littleGnAl sounds good, i'll do a squash of these commits without the other parts to avoid conflicts

**Review by littleGnAl — APPROVED** [review]

**Comment by littleGnAl** [comment]
LGTM

**Pull request merged**